File: dotnet/Kata/External/ConversionRateApi.cs
namespace Kata.External;

public class ConversionRateApi
{
    public double GetRate(CurrencyIsoCode source, CurrencyIsoCode target)
    {
        Console.WriteLine($"Getting rate from {source.GetDescription()} to {target.GetDescription()}");
        throw new ArgumentException(
            "401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?");
    }
}

File: kotlin/src/main/kotlin/org/katas/external/conversionRateApi/ConversionRateApi.kt
package org.katas.external.conversionRateApi

import org.katas.external.CurrencyIsoCode

class ConversionRateApi {
    fun getRate(source: CurrencyIsoCode, target: CurrencyIsoCode): Double {
        throw RuntimeException("401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?")
    }
}